junit5-java9-engine/src/main/java/ice/cream/Machine.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2015-2017 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package ice.cream;
+
+import org.junit.platform.engine.EngineDiscoveryRequest;
+import org.junit.platform.engine.EngineExecutionListener;
+import org.junit.platform.engine.ExecutionRequest;
+import org.junit.platform.engine.TestDescriptor;
+import org.junit.platform.engine.TestEngine;
+import org.junit.platform.engine.TestExecutionResult;
+import org.junit.platform.engine.UniqueId;
+import org.junit.platform.engine.support.descriptor.EngineDescriptor;
+
+public class Machine implements TestEngine {
+
+  private FlavorDescriptor sample;
```

Comment:
We should not encourage engine authors to store descriptors in instance variables.

Reply:
True. Replaced by generating 3 flavors while discovering and "parsing" them back when executing.